query-engine/connectors/query-connector/src/filter/scalar.rs

```diff
@@ -576,3 +576,195 @@ impl ScalarCompare for ModelProjection {
         })
     }
 }
+
+impl ScalarCompare for FieldSelection {
+    /// Field is in a given value
+    fn is_in<T>(&self, values: Vec<T>) -> Filter
+    where
+        T: Into<PrismaValue>,
+    {
+        Filter::from(ScalarFilter {
+            projection: ScalarProjection::Compound(self.as_scalar_fields().expect("Todo composites in filters.")),
+            condition: ScalarCondition::In(values.into_iter().map(|i| i.into()).collect()),
```

Comment:
Not really, because of the generic bound `T: Into<PrismaValue>`  you need to convert no matter what, so pusing it out to a function doesn't do you any good. The only thing that can be done better here that is relative the the gain is `values.into_iter().map(Into::into).collect()`.